nrkno/sofie-core
# fix(Client): Piece-dependency for expectedDuration calculation causes the Diff to be inconsistent (SOFIE-3004)

## About the Contributor

This PR is posted on behalf of the NRK.


## Type of Contribution

This is a:

Bug fix


## Current Behavior

Calculating the Diff requires knowing all Part's expectedDurations, that means that in order to calculate the Diff, one needs to have all the Pieces for a Playlist loaded. The GUI trims down the amount of data it needs in local cache ("minimongo") in order to maximize performance, and as such, will only have some Pieces loaded at any given time. The Presenters' screen for example will only have the Pieces for the current and next Part loaded. This causes the Diff to be different on the Presenters' screen and in the Rundown View. The Diff in the Rundown View changes as the user scrolls around the page.


## New Behavior

The expectedDuration calculation does not consider Pieces prerolls. This will introduce minimal (in real-life scenarios) innacuracies in the calculation, but at least will make the Diff counter consistent.


## Testing Instructions

* Open Rundown View
* Open Presenter screen
* Make sure if the Diff is consistent


## Time Frame

* This Bug Fix is critical for us, please review and merge it as soon as possible. 


## Other Information



## Status
